:memo: Rework HTTPS page (#2888)
platformsh/platformsh-docs: docs/src/define-routes/https.md

configure your own [third-party TLS certificates](../domains/steps/tls.md).

### Let's Encrypt limitations

When you use the Let's Encrypt [TLS certificates](#tls-certificates) provided by Platform.sh,
the following limitations apply.  

{{% lets_encrypt_limitations %}}

If you need more hostnames, you can obtain additional certificates
or a wildcard certificate from a [third-party issuer](../domains/steps/tls.md).
Alternatively, consider splitting your project up into multiple Platform.sh projects.

### Certificate renewals

When you use the [TLS certificates](#tls-certificates) provided by Platform.sh,
certificate renewals are automatic.
They trigger a redeployment of your environment.
During this redeployment, required security and system upgrades are applied to your containers
and only the [`post-deploy` hook is run](../create-apps/hooks/hooks-comparison.md).

Renewals take seconds unless upgrades are available for your containers.
If upgrades are available, your containers are rebuilt, which lengthens the process.

## Enable HTTPS

To enable HTTPS, add a routing configuration similar to the following:

```yaml {location=".platform/routes.yaml"}
"https://{default}/":

>    type: upstream
>    upstream: "app:http"

Reviewer: :clipboard: 
```suggestion
    type: upstream
    upstream: "app:http"
```

"https://www.{default}/":

>    type: redirect
>    to: "https://{default}/"

Reviewer: :clipboard: 
```suggestion
    type: redirect
    to: "https://{default}/"
```

```

All traffic to your domain is then sent to your app.
The `www` subdomain redirects to the [default domain](../define-routes/_index.md#default).
This also includes redirecting requests from HTTP to HTTPS.

For more information, see how to [define routes](../define-routes/_index.md).

## Optional: Configure TLS connections

When you [specify HTTPS routes](#enable-https),
you can use the `tls` setting to further configure your TLS connections.

### Enforce TLS 1.3

Although you can still use TLS 1.2, TLS 1.3 is faster and more secure.
To instruct your web server to automatically reject TLS 1.2 connections,
enforce TLS 1.3 using the `min_version` setting:

```yaml {location=".platform/routes.yaml"}
tls:

>    min_version: TLSv1.3

Reviewer: :clipboard: 
```suggestion
    min_version: TLSv1.3
```

```

Note that TLS versions older than 1.2 are deprecated and are rejected by default.

### Enable HTTP Strict Transport Security (HSTS)

[HSTS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) forces clients to always communicate with your site over HTTPS.
To enable HSTS, use `strict_transport_security` in a configuration similar to the following:

```yaml {location=".platform/routes.yaml"}
tls:

>    strict_transport_security:
>        enabled: true
>        include_subdomains: true
>        preload: true

Reviewer: :clipboard: 
```suggestion
    strict_transport_security:
        enabled: true
        include_subdomains: true
        preload: true
```

| `enabled`            | `boolean` | `null`  | If set to `true`, HSTS is enabled for 1 year. If set to `false`, other properties are ignored. |
| `include_subdomains` | `boolean` | `false` | To specify whether HSTS applies to all subdomains.                                             |
| `preload`            | `boolean` | `false` | To add your website to the [HSTS preload list](https://hstspreload.org/). Thanks to this list, most browsers are informed that your site requires HSTS before an HSTS header response is even issued. |

Note that when you enable or disable HSTS, the entire domain is affected.
Make sure you only add the HSTS configuration to a single route.
Having different routes with conflicting HSTS configurations can cause issues.

### Enable client-authenticated TLS

Standard TLS connections are useful to verify the identity of web servers and their certificates.
But you can also instruct your web server to verify the identity of clients and their certificates.
This allows you to restrict access to trusted users.

To do so, enable client-authenticated TLS by adding the following configuration:

```yaml {location=".platform/routes.yaml"}
tls:

>    client_authentication: "require"

Reviewer: :clipboard: 
```suggestion
    client_authentication: "require"
```

```

By default, all valid TLS certificates issued by a legitimate certificate authority are accepted.
But you can instruct your web server to only accept TLS certificates issued by specific or even custom certificate authorities.

To do so, add a configuration similar to the following:

```yaml {location=".platform/routes.yaml"}
tls:

>    client_authentication: "require"
>    client_certificate_authorities:
>        - !include
>            type: string
>            path: root-ca1.crt
>        - !include
>            type: string
>            path: root-ca2.crt

Reviewer: :clipboard: 
```suggestion
    client_authentication: "require"
    client_certificate_authorities:
        - !include
            type: string
            path: root-ca1.crt
        - !include
            type: string
            path: root-ca2.crt
```

```

In this case, the certificate files are resolved relative to the `.platform` directory.
Alternatively, you can specify the certificates inline in the file:

```yaml {location=".platform/routes.yaml"}
tls:

>    client_authentication: "require"
>    client_certificate_authorities:
>        - |
>            -----BEGIN CERTIFICATE-----
>            ### Several lines of characters here ###
>            -----END CERTIFICATE-----
>        - |
>            -----BEGIN CERTIFICATE-----
>            ### Several lines of different characters here ###
>            -----END CERTIFICATE-----

Reviewer: :clipboard: 
```suggestion
    client_authentication: "require"
    client_certificate_authorities:
        - |
            -----BEGIN CERTIFICATE-----
            ### Several lines of characters here ###
            -----END CERTIFICATE-----
        - |
            -----BEGIN CERTIFICATE-----
            ### Several lines of different characters here ###
            -----END CERTIFICATE-----
```

---
title: HTTPS
weight: 3
description: Learn how to enable HTTPS for your site and configure TLS connections.

> sidebarTitle: "HTTPS"

Reviewer: :pick: This is the same as the `title` and so unnecessary
```suggestion
```

> The sub-properties of `strict_transport_security` are:

Reviewer: :hammer_and_wrench: 
```suggestion
The following table presents the possible properties for `strict_transport_security`:
```

            -----BEGIN CERTIFICATE-----

>             ### Several lines of different random characters here ###

Reviewer: :hammer_and_wrench: Here and above, I'd avoid the use of the word `random`, because they're not really random, just generated by the certificate generator.
